pdp/static/js/pdp_auth.js

```diff
@@ -41,117 +24,132 @@ window.pdp = (function (my, $) {
-                success: function () {
-                    button.prop("loggedIn", false);
-                    button.html("Login with OpenID");
-                    alert("Logout successful.");
-                },
-                // On 401, we are now unauthorized, redirect somewhere?
-                error: function () {
-                    alert("We're not authorized! I'm not sure what to do. Abort! Abort!");
+        hello.on('auth.login', function(auth) {
+            hello( auth.network ).api( '/me' ).then( function(r){
+                user = r;
+
+                if (pdp.validateEmail(user.email)) {
+                    pdp.login(user.email, auth.network);
+                } else {
+                    oauthErrorDialog.dialog("open");
                 }
             });
-            return false;
-        }
+        });
 
-        function startSignup(evt) {
-            var name = $("select[name='openid-provider'] option:selected").html(),
-                url = signupUrls[name];
-            window.open(url);
-        }
+        hello.on('auth.logout', function(auth) {
+            user = undefined;
+        });
 
-        /*jslint unparam: true*/
-        function doSubmitAfterLogin(evt) {
-            $("#filter").submit();
-            $("#do-login").unbind("click");
-            $("#do-login").click(startLogin); // clear doSubmit as the login onSuccess event handler
-        }
+        /* FIXME: check if session has registered email already. Since hellojs
+        automatically logs back in if the session is valid, this must only be
+        done if we know hellojs does not have a valid session. */
 
-        function showLogin(evt) {
-            form.dialog("open");
-            return false;
-        }
+        return loginButton;
+    };
 
-        function toggleLogin(evt) {
-            if (button.prop("loggedIn")) {
-                // Log out
-                doLogout(evt);
-            } else {
-                // Log in
-                showLogin(evt);
+    my.login = function (email, network) {
+
+        // Register login through button
+        var loginButton = $("#login-button")
+        loginButton.prop("loggedIn", true);
+        loginButton.hide();
+
+        // Create logout button
+        var link = document.createElement("a");
+        link.id = "logout-button"
+        link.appendChild(document.createTextNode("Logout as " + email));
+        $(link).click(function() {
+            pdp.logout(network);
+        });
+        document.getElementById("login-button").parentElement.appendChild(link);
+
+        // Log into the server
+        $.ajax({
+            type: "POST",
+            url: pdp.app_root + '/user/login',
+            data: {
+                "email": email
             }
+        })
+        // FIXME Add failure handler
+    }
+
+    my.logout = function(network) {
+
+        // oauth logout
+        if (network) {
```

Comment:
Soooo, can one call `logout()` with no network to force a request to /user/logout and show the loginButton? Or what happens if `logout()` is called with no network defined?